[SETTING/#2] 어플리케이션 이관

yummygyudon opened the pull request:
## Related Issue 🚀
- cloesd #1 

## Work Description ✏️
- [Legacy 인증 프로젝트](https://github.com/sopt-makers/sopt-auth-backend-legacy)에 설정되어 있는 기본 어플리케이션 설정을 이관합니다.
- 멀티모듈에서 모놀리식 프로젝트로 변경하면서 추가 변경 및 반영, 그리고 최대한 개선했습니다.

## PR Point 📸
### 1. Application Profile 별 YAML 파일 분리

Profile 별로 어플리케이션 설정 파일을 분리했습니다.(prod/dev/local/test)
- `application-{profile}.yaml` 명명 규칙으로 설정했습니다.

<br/>

### 2. 설정 파일 세분화
현재 4가지의 profile로 분리하여 운영하는 상황에서 단일 설정 파일 내에 `---`를 통해 profile을 분리하여 작성할 경우, 
지나치게 많은 반복적인 구조의 내용들이 작성되어 가독성을 해칠 것이라고 판단했습니다.

- `spring.config.import` 를 통해 필요한 yaml을 주입받아 사용하도록 했습니다.
- 각 profile에 특화된 내용만 작성되도록 했습니다.

#### 이유
- 가독성 개선
- 독립적인 관리 및 관리 용이성 제고

<br/>

### 3. ConfigurationProperties 지속 활용
이전 프로젝트와 동일하게 Record 타입을 활용한 `@ConfigurationProperties`를 지속적으로 활용했습니다.
단, yaml의 depth를 개편하여`prefix`를 통해 분리된 Property 객체로 사용될 수 있도록 했습니다.
(_ex. MakersProperty :: prefix="external.makers", GabiaProperty :: prefix="external.gabia"_)

yummygyudon committed:
[SETTING] 기본 어플리케이션 프로젝트 설정

- 대부분의 프로젝트 의존성 및 설정값은 [Legacy 인증 프로젝트](https://github.com/sopt-makers/sopt-auth-backend-legacy)를 참고해주세요.
- application을 Profile을 기준으로 분리했습니다.

yummygyudon committed:
[FEAT] 설정 파일 Property 객체 구현

- 어플리케이션 설정파일(`yaml`) 값을 주입 받는 클래스입니다.
- `@ConfigurationProperties` 어노테이션을 사용했습니다.
  - Prefix 설정을 통해 관심사를 분리하여 각 객체를 구현했습니다.

yummygyudon committed:
[FEAT] 기본적인 ResponseData,에러/성공 HTTP Status, 예외 계층 구조 구현

다형성과 접근제어자, 제네릭을 통해 추후 Type-Safe한 확장이 용이하도록 계층 구조를 구축했습니다.
- 추상 클래스, 인터페이스를 통해 객체 생성 방지 및 슈퍼 타입 활용을 통해 구현했습니다.

* 이전 커밋에서 놓쳤던 Property 클래스(`AuthProperty`)를 포함하여 커밋했습니다.

yummygyudon committed:
[FEAT] Spring 컨테이너 Configuration 객체 및 상수(Constant) 객체 구현

- ApplicationConfig

- JPA Config
- Spring Security Config
  - JWT 발급을 위해 필요한 Provider까지 구현했습니다.
  - 설정에 필요한 ResponseUtil 클래스와 Filter 객체까지 구현했습니다.
- Constant
  - SystemConstant : 시스템적으로 전역적인 설정 값으로 사용되는 요소 상수 정의

yummygyudon committed:
[CHORE] 패키지 구조 표현을 위해 임시 및 예시 클래스 구현

* 이전 커밋에 누락된 `SystemConstant` 클래스 이력을 추가했습니다.
* 기본 어플리케이션 Test 클래스 `AuthenticationApplicationTests`도 추가했습니다.

hyunw9 reviewed (commented):
확인했습니다 ~ 구조 깔끔하니 좋은것 같습니다. 

고생하셨습니다 !

sung-silver commented on .github/release-drafter-config.yml:
```diff
@@ -0,0 +1,37 @@
+name-template: 'v$RESOLVED_VERSION'
+tag-template: 'v$RESOLVED_VERSION'
+categories:
```
라벨 확인했습니다! 다만 현재 pr에서 setting, feature 라벨 두 개가 달려있는데 둘 중 하나만 달아주셔야 release drafter에서 제대로 인식하게 됩니다! 이관 작업이니 feature를 지우고 setting을 남기면 좋을 것 같습니다!

yummygyudon replied:
확인했습니당!!

sung-silver commented on src/main/java/sopt/makers/authentication/support/constant/SystemConstant.java:
```diff
@@ -0,0 +1,17 @@
+
+  public static final String PATTERN_ALL = "/**";
+  public static final String PATTERN_ERROR_PATH = "/error";
+  public static final String PATTERN_SWAGGER = "/swagger-ui/**";
```
혹시 저희 swagger도 사용하나요?!

hyunw9 replied:
아 맞네요 저번에 레스트독 쓴다고 했던거 같긴 했는데..

yummygyudon replied:
아핫!! 지우도록 하겠습니다!

sung-silver reviewed (changes requested):
리뷰 확인부탁드립니다!

yummygyudon committed:
[STYLE] DTO 생성 컨벤션 변경

DTO 객체 관리 용이성 및 명확한 책임과 관심사 분리를 위해 SubClass 개념을 도입했습니다.
- 상위 클래스 명을 명사형으로 선언한 후, 서브 클래스 명으로 행위를 명시하도록 했습니다.
- 상위 클래스의 경우, `{Domain}Request` / `{Domain}Response` 형식입니다.
- 서브 클래스의 경우, `{Action}{DetailContent}` 형식입니다.

yummygyudon committed:
[CHORE] Security permit all 패턴 중 미사용 URL 포함 제거

[본 리뷰](https://github.com/sopt-makers/sopt-auth-backend/pull/2#discussion_r1827149162)에 대한 내용 반영입니다.
